skimage/registration/_lddmm.py

```diff
@@ -0,0 +1,1247 @@
+
+    def register(self):
+        """
+        Register the template to the target using the current state of the attributes.
+        
+        Return a dictionary of relevant quantities most notably including the transformations:
+            phi_inv_affine_inv is the position field that maps the template to the target.
+            affine_phi is the position field that maps the target to the template.
+        """
+
+        # Iteratively perform each step of the registration.
+        for iteration in range(self.num_iterations):
+            # If self.track_progress_every_n > 0, print progress updates every 10 iterations.
+            if self.track_progress_every_n > 0 and not iteration % self.track_progress_every_n:
+                print(
+                    f"Progress: iteration {iteration}/{self.num_iterations}"
+                    f"{' rigid' if iteration < self.num_rigid_affine_iterations else ''}"
+                    f"{' affine only' if iteration < self.num_affine_only_iterations else ' affine and deformative'}."
+                )
+
+            # Forward pass: apply transforms to the template and compute the costs.
+
+            # Compute position_field from velocity_fields.
+            self._update_and_apply_position_field()
+            # Contrast transform the deformed_template.
+            self._apply_contrast_map()
+            # Compute weights. 
+            # This is the expectation step of the expectation maximization algorithm.
+            if self.check_artifacts and iteration % 1 == 0: self._compute_weights()
+            # Compute cost.
+            self._compute_cost()
+
+            # Backward pass: update contrast map, affine, & velocity_fields.
+            
+            # Compute contrast map.
+            self._compute_contrast_map()
+
+            # Compute affine gradient.
+            affine_inv_gradient = self._compute_affine_inv_gradient()
+            # Compute velocity_fields gradient.
+            if iteration >= self.num_affine_only_iterations: velocity_fields_gradients = self._compute_velocity_fields_gradients()
+            # Update affine.
+            self._update_affine(affine_inv_gradient, iteration)
+            # Update velocity_fields.
+            if iteration >= self.num_affine_only_iterations: self._update_velocity_fields(velocity_fields_gradients)
+
+        # Compute affine_phi in case there were only affine-only iterations.
+        self._compute_affine_phi()
+
+        # Optionally display useful plots for calibrating the registration parameters.
+        if self.calibrate:
+            self._generate_calibration_plots()
+        
+        # Note: the user-level lddmm_transform_image function relies on many of these specific outputs with these specific keys to function. 
+        # ----> Check the lddmm_transform_image function signature before adjusting these outputs.
```

Comment:
I've attempted to tackle this with 3 changes - 

- The signature now has only 7 arguments with the rest relegated into **kwargs,
- the output is now an organized namedtuple rather than a dictionary bucket of values,
- and the apply-functions have been deprecated in favor of scipy.ndimage.map_coordinates.